R/checks.R

```diff
@@ -533,3 +533,19 @@ check_gp_failure <- function(current, prev) {
   # return prev model
   prev
 }
+
+check_no_tuning <- function(x) {
+  tune_param <- tune_args(x)
+  num_param <- nrow(tune_param)
+  if (num_param == 0) {
+    return(invisible(FALSE))
+  }
+  srcs <- unique(tune_param$source)
+  num_srcs <- length(srcs)
+  srcs <- paste0(srcs, collapse = " and ")
+  msg_1 <- cli::pluralize("{num_param} argument{?s} {?has/have} been tagged for tuning in {?this component/these components}: {srcs}. ")
+  msg_2 <- "Please use one of the tuning functions (e.g. `tune_grid()`) to optimize them."
+  msg <- paste(msg_1, msg_2, sep = "\n")
+  rlang::abort(msg)
```

Comment:
```suggestion
  rlang::abort(msg, call = NULL)
```

Reply:
Since this is a user-facing error, we can hide the source line/helper name here:

``` r
boop <- function() {
  rlang::abort("woah!")
}

bop <- function() {
  rlang::abort("woah!", call = NULL)
}

boop()
#> Error in `boop()`:
#> ! woah!

bop()
#> Error:
#> ! woah!
```

<sup>Created on 2022-06-04 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

It looks like snapshots may not need to be updated with this change.

Reply:
One thing we might want to think about with error messages is to move away from bespoke `\n` because the rlang / cli infrastructure handles that better relative to the console window. Instead of pasting with `\n`, we can use a vector that handles it automatically:

``` r
rlang::abort(c("Oh no!", "This is a real problem"))
#> Error:
#> ! Oh no!
#> • This is a real problem
```

<sup>Created on 2022-06-05 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Reply:
This is being fixed in #513 where we get all of the lines:


> res_roles <- role_bp_wflow %>% fit_resamples(rs)
x Fold01: preprocessor 1/1, model 1/1 (predictions):
  Error:
  ! The following cols are missing from `new_data`: date.
  ℹ There are also non-standard recipe roles for the column(s).
  ℹ See `?update_role` for more information on how use non-standard recipe roles during prediction.
x Fold02: preprocessor 1/1, model 1/1 (predictions):
  Error:
  ! The following cols are missing from `new_data`: date.
  ℹ There are also non-standard recipe roles for the column(s).
  ℹ See `?update_role` for more information on how use non-standard recipe roles during prediction.
```

and so on.